src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/SearchQueryOperation.java

```diff
@@ -87,27 +90,57 @@ public SearchResultView execute(final String view,
         return mergeDataToSearchResultOperation.execute(caseType.get(), searchResult, draftsAndCases, draftResultError);
     }
 
-    private SearchResult getSearchResultDefinition(final CaseTypeDefinition caseTypeDefinition, final String view) {
-        if (WORKBASKET.equalsIgnoreCase(view)) {
-            return uiDefinitionRepository.getWorkBasketResult(caseTypeDefinition.getId());
+    public SearchResult getSearchResultDefinition(final CaseTypeDefinition caseTypeDefinition, final String useCase) {
+        final String caseTypeId = caseTypeDefinition.getId();
+        if (Strings.isNullOrEmpty(useCase)) {
+            return buildSearchResultFromCaseFields(caseTypeDefinition);
+        }
+        // TODO: Once all *ResultFields tabs are merged, remove switch statement and always call default method
+        switch (useCase) {
+            case WORKBASKET:
+                return uiDefinitionRepository.getWorkBasketResult(caseTypeId);
+            case SEARCH:
+                return uiDefinitionRepository.getSearchResult(caseTypeId);
+            default:
+                return uiDefinitionRepository.getSearchCasesResult(caseTypeId, useCase);
         }
-        return uiDefinitionRepository.getSearchResult(caseTypeDefinition.getId());
     }
 
-    private void addSortOrderFields(MetaData metadata,SearchResult searchResult) {
-        List<SortOrderField> sortOrders = getSortOrders(searchResult);
-        metadata.setSortOrderFields(sortOrders);
+    public List<SortOrderField> getSortOrders(CaseTypeDefinition caseType, String useCase) {
+        return getSortOrders(getSearchResultDefinition(caseType, useCase));
     }
 
     private List<SortOrderField> getSortOrders(SearchResult searchResult) {
         return Arrays.stream(searchResult.getFields())
-            .filter(searchResultField -> hasSortField(searchResultField))
-            .filter(searchResultField -> filterByRole(searchResultField))
+            .filter(this::hasSortField)
+            .filter(this::filterByRole)
             .sorted(Comparator.comparing(srf -> srf.getSortOrder().getPriority()))
             .map(this::toSortOrderField)
             .collect(Collectors.toList());
     }
 
+    private SearchResult buildSearchResultFromCaseFields(final CaseTypeDefinition caseTypeDefinition) {
```

Comment:
minor: 
I've noticed that the "getSearchResultDefinition" is called multiple times, which will call this code multiple
times. Given that the performance of the search operation is important and this is potentially slow for big case types, we could consider refactoring to have it called only once if not a big change

Reply:
Made some changes in the generator to reduce the number since we can assume it'll always be a single case type for all items currently 👍